_gitbook/syntax_and_semantics/macros/macro_methods.md

```diff
@@ -20,3 +20,19 @@ person.instance_vars_names #=> ["name", "age"]
 
 Note that in the case of macro defs you need to specify the return type.
 
+Arguments in macro defs are evaluated in running time, not a compile time:
```

Comment:
Maybe we should just include distinction a bit more explicit here by repeating the behaviour for macros, how about:

> In macro definitions arguments are passed as their AST nodes, giving you access to them in macro expansions (`{{some_macro_argument}}`). However that is not true for macro defs, here the argument list is that of the method generated by the macro def, you cannot not access their compile time value.

>  ```cr
class Object
  macro def has_instance_var?(name) : Bool
    # We cannot access name inside the macro expansion here, instead we need to use the
    # macro language to construct an array and do the inclusion check at runtime.
    return {{ @type.instance_vars.map &.name.stringify }}.includes? name
  end
end